tests/test_models/test_simple_models.py

```diff
@@ -39,29 +39,45 @@ def df():
 def test_simple_model_forecaster_run(simple_df, model):
     sma_model = model()
     sma_model.fit(simple_df)
-    future_ts = simple_df.make_future(future_steps=7)
-    res = sma_model.forecast(future_ts)
+    future_ts = simple_df.make_future(future_steps=7, tail_steps=sma_model.context_size)
+    res = sma_model.forecast(future_ts, prediction_size=7)
     res = res.to_pandas(flatten=True)
     assert not res.isnull().values.any()
     assert len(res) == 14
 
 
+def test_simple_model_forecaster_fail(simple_df):
+    sma_model = SeasonalMovingAverageModel(window=1000, seasonality=7)
+    sma_model.fit(simple_df)
+    future_ts = simple_df.make_future(future_steps=7, tail_steps=sma_model.context_size)
+    with pytest.raises(ValueError, match="Given context isn't big enough"):
+        _ = sma_model.forecast(future_ts, prediction_size=7)
+
+
 @pytest.mark.parametrize("model", [DeadlineMovingAverageModel])
 def test_deadline_model_forecaster_run(simple_df, model):
     model = model(window=1)
     model.fit(simple_df)
-    future_ts = simple_df.make_future(future_steps=7)
-    res = model.forecast(future_ts)
+    future_ts = simple_df.make_future(future_steps=7, tail_steps=model.context_size)
+    res = model.forecast(future_ts, prediction_size=7)
     res = res.to_pandas(flatten=True)
     assert not res.isnull().values.any()
     assert len(res) == 14
 
 
+def test_sdeadline_model_forecaster_fail(simple_df):
```

Comment:
May be also a test for `_get_context_begining`

Reply:
Added.